ompi/mca/pml/ucx/pml_ucx_request.h

```diff
@@ -186,15 +186,19 @@ static inline void mca_pml_ucx_set_recv_status(ompi_status_public_t* mpi_status,
     } else {
         mpi_status->MPI_ERROR = MPI_ERR_INTERN;
     }
+
+    return mpi_status->MPI_ERROR;
 }
 
-static inline void mca_pml_ucx_set_recv_status_safe(ompi_status_public_t* mpi_status,
-                                                    ucs_status_t ucp_status,
-                                                    const ucp_tag_recv_info_t *info)
+static inline int mca_pml_ucx_set_recv_status_safe(ompi_status_public_t* _mpi_status,
+                                                   ucs_status_t ucp_status,
+                                                   const ucp_tag_recv_info_t *info)
 {
-    if (mpi_status != MPI_STATUS_IGNORE) {
-        mca_pml_ucx_set_recv_status(mpi_status, ucp_status, info);
-    }
+    ompi_status_public_t _local_status;
+    ompi_status_public_t *mpi_status = (_mpi_status != MPI_STATUS_IGNORE) ?
+                                       _mpi_status : &_local_status;
+
+    return mca_pml_ucx_set_recv_status(mpi_status, ucp_status, info);
```

Comment:
maybe avoid setting temp mpi_status field if not needed
```
if (mpi_status != MPI_STATUS_IGNORE) {
    return mca_pml_ucx_set_recv_status(mpi_status, ucp_status, info)
} else if (status == UCS_OK || status == CANCELED) {
   return UCS_OK;
} else if (status == TRUNCATED)
  reutrn ERR_TRUNCATE
} else {
 return ERR_INTERN
}
```

Reply:
updated